### PR TITLE
Updated ws definitions to 7.2

### PR DESCRIPTION
**Pull request by ghost**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/websockets/ws/releases
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by teidesu** [comment]
it does exist on master, welp
https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/ws

File: types/ws/index.d.ts [hunk 1]
```diff
@@ -230,15 +234,18 @@ declare namespace WebSocket {
         on(event: 'connection', cb: (this: WebSocket, socket: WebSocket, request: http.IncomingMessage) => void): this;
         on(event: 'error', cb: (this: WebSocket, error: Error) => void): this;
         on(event: 'headers', cb: (this: WebSocket, headers: string[], request: http.IncomingMessage) => void): this;
-        on(event: 'listening', cb: (this: WebSocket) => void): this;
+        on(event: 'closed' | 'listening', cb: (this: WebSocket) => void): this;
```
Comment (reduckted) [inline types/ws/index.d.ts]: I think the event name is actually `close`. https://github.com/websockets/ws/blob/0a612364e69fc07624b8010c6873f7766743a8e3/doc/ws.md#event-close

**Review by reduckted — CHANGES_REQUESTED** [review]
Apart from the `close` event, everything else looks good.

**Review by reduckted — APPROVED** [review]

**Comment by hellola** [comment]
Is there a reason the option for constructing with a `url.Url` type has been removed?

**Comment by reduckted** [comment]
> @hellola Is there a reason the option for constructing with a url.Url type has been removed?

Hmm, probably some confusion around this in the release notes:

https://github.com/websockets/ws/releases/tag/7.0.0

> Dropped support for `url.Url` instances in the `WebSocket` constructor
([692d7b4](https://github.com/websockets/ws/commit/692d7b47624c5392d78fe4ea1351b5daa17bad94)).

Looking at the code changes though, it seems like support for `url.Url` was dropped, but support for `url.URL` still exists.

**Comment by reduckted** [comment]
@hellola Looks like #41905 will fix this.

**Pull request merged**